core/src/main/java/org/pentaho/di/connections/vfs/provider/ConnectionFileSystem.java

```diff
@@ -65,12 +65,17 @@ public static String getUrl( AbstractFileName abstractFileName, ConnectionDetail
     String url = null;
 
     if ( vfsConnectionDetails != null ) {
-      String domain = vfsConnectionDetails.getDomain();
-      if ( !domain.equals( "" ) ) {
-        domain = "/" + domain;
+      if ( "smb".equals( vfsConnectionDetails.getType().toLowerCase() ) ) {
+        url = abstractFileName.getURI().replaceFirst( "pvfs:", "smb:" );
```

Comment:
Can we store smb: in some common constant and use it instead of having the string re-typed in every class that uses it?